app/raw/[...slug]/route.ts

```diff
@@ -10,12 +11,41 @@ export async function GET(_: Request, { params }: { params: Promise<{ slug: stri
     // 1. They have the `.mdx` extension converted to `.md`
     // 2. They have the `docs` slug prefix removed (removed from the passed github file url)
     const slug = (await params).slug.map((s) => s.replace(".md", ".mdx"));
-    const filePath = path.join(process.cwd(), "docs", slug.join("/"));
-    const { content, title, description } = await getRawDocContent(filePath);
-    const merged = `# ${title}\n\n${description}\n\n${content}`;
-    const filename = slug.pop() || "index.md";
 
-    return new NextResponse(merged, {
+    // Check if this is an API reference page
+    const isApiReferencePage = slug.join("/").includes("reference/endpoints/");
+    const isApiReferenceRootPage = slug.length === 2 && slug[0] === "reference" && slug[1] === "endpoints";
+
+    let content: string;
+    let filename: string;
+
+    if (isApiReferencePage && !isApiReferenceRootPage) {
+      // For API pages, serve pre-generated markdown
+      const lastSlug = slug[slug.length - 1];
+      if (!lastSlug) {
```

Comment:
This section makes me slightly nervous? I have no idea _how_ but someone could find a vulnerability here. What if we just do a little path validation, and ensure we always have something pretty simple that we know we'll always have?

```ts
if (!/^[a-zA-Z0-9._-]+\.mdx$/.test(lastSlug)) {
  return new NextResponse("Not found", { status: 404 });
}
```

No slashes in there or dots so that we don't accidentally serve the wrong files?